src/Test/TestCombatMods.java

```diff
@@ -158,4 +163,84 @@ private boolean testRetribution()
 
     return testPassed;
   }
+
+  private boolean testDamageDealtToIncomeConverter()
+  {
+    boolean testPassed = true; // Assume nothin's busted
+    DamageDealtToIncomeConverter ddtic = DamageDealtToIncomeConverter.instance(testGame, DamageDealtToIncomeConverter.class);
+
+    Unit aa = addUnit(testMap, cinder, UnitModel.SURFACE_TO_AIR, 7, 3);
+    // juice aa to be extra angry and get overkills
+    aa.alterHP(10, true);
+    aa.addUnitModifier(new CODamageModifier(42));
+
+    int currentFundsReturn = 0;
+    // Get lots of free money, and make sure we get the right amount
+    for( int i = 0; i < 100; ++i )
+    {
+      currentFundsReturn += i;
+      ddtic.startTracking(cinder, i);
+      cinder.money = 0;
+
+      // give aa ammo
+      aa.isTurnOver = false;
+      ResupplyEvent event = new ResupplyEvent(null, aa);
+      event.performEvent( testGame.gameMap );
+      GameEventListener.publishEvent(event, testGame);
+
+      // Add the victim, and yeet him
+      addUnit(testMap, venge, UnitModel.TROOP, 7, 4);
+      performGameAction(new BattleLifecycle.BattleAction(testMap, aa, Utils.findShortestPath(aa, 7, 3, testMap), 7, 4), testGame);
+
+      testPassed &= validate(cinder.money == currentFundsReturn * 1000, "    Expected to make "+currentFundsReturn*1000+", but got "+cinder.money+" instead.");
+    }
+    // Remove those amounts, and make sure that works
+    for( int i = 0; i < 55; ++i )
+    {
+      currentFundsReturn -= i;
```

Comment:
Isn't this going to check the same cases as the previous loop, just in reverse as `currentFundsReturn` goes back to `0`?

Reply:
Yes, but it's removing them in a different order than they were added.